manager/manager/typeresolver.go

```diff
@@ -92,18 +93,27 @@ func (tr *typeResolver) ResolveTypes(config *common.Configuration, imports []*co
 	}
```

Comment:
This code has gotten pretty gnarly with the deep nested ifs and fors. Can you try to break it up a bit?